backend/api/src/lib.rs

```diff
@@ -12,10 +12,10 @@ pub mod mutation;
 pub mod query;
```

Comment:
I still see `pub mod` scattered around here. Can these not be `pub(crate)`? Shouldn't they be?

Reply:
Unfortunately not, at least right now. The only thing that needs to be public is the `root_node` function. But that returns a type and that type also needs to be public because it is used in a public function. One could use existential types (let `root_node` return `impl ...`, but the trait we need to use there seems to be highly complicated and would be really terrible to read :D 

So I kept it as is, meaning that `RootNode` has to be public, which includes `Query`, `Mutation` and `Subscription`, which all in turn have to be public as well. Their modules don't need to be public per se, but I simply didn't change it yet. And ok, it is possible to have public APIs expose non-public types.

In short: there are reasons, but we could change it. We will change stuff in the future anyway. It's not that bad right now.

Reply:
Fair enough.